packages/components/MenuButton/src/MenuButton.tsx

```diff
@@ -31,6 +31,36 @@ export const MenuButton = compose<MenuButtonType>({
       setShowContextualMenu(!showContextualMenu);
     }, [showContextualMenu, setShowContextualMenu]);
 
+    const menuItemsUpdated = menuItems.map((item) => {
+      if (item.hasSubmenu) {
+        const [showSubmenu, setShowSubmenu] = useState(false);
+        const [isSubmenuVisible, setIsSubmenuVisible] = React.useState(false);
```

Comment:
It looks like this isn't used, and that showSubmenu can take care of whatever state this is tracking

Reply:
you're right, this one is not needed here, fixed

Reply:
But onShow function is required. If I remove it, there will be an error, that onShow is not a fucntion.

Reply:
We made it optional!